ynvest_tube/app/src/main/java/com/app/ynvest_tube/fragments/BalanceBarFragment.kt

```diff
@@ -9,26 +9,23 @@ import android.widget.TextView
 import android.widget.Toast
 import com.app.ynvest_tube.R
 import com.app.ynvest_tube.model.User
+import com.app.ynvest_tube.refresher.DataRefresher
+import com.app.ynvest_tube.refresher.UserSubscriber
 import com.app.ynvest_tube.repository.Repository
 
 class BalanceBarFragment : Fragment() {
 
-    private val repository = Repository()
     private lateinit var fragmentView: View
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         fragmentView = inflater.inflate(R.layout.fragment_balance_bar, container, false)
-        refresh()
+        DataRefresher.userSubscribers["barView"] = UserSubscriber(::userObtained)
         return fragmentView
     }
 
-    fun refresh() {
-        repository.getUser(::userObtained, ::requestFailed)
```

Comment:
I think there is no need to refresh this since we have parodic refresh. I know this way it could refresh faster, but it is quite complicated case and we would need to talk about it.